esmvalcore/preprocessor/_multimodel.py

```diff
@@ -618,11 +618,16 @@ def _multiproduct_statistics(products,
         ignore_scalar_coords=ignore_scalar_coords,
     )
     statistics_products = set()
+
+    # Create copies of products to avoid recursively adding ancestors
+    copied_products = {p.copy_provenance() for p in products}
```

Comment:
These copies are not needed, since `statistics_products` are never derived from other `statistics_products`, only from the original input `products` to the function.